src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -663,35 +665,41 @@ protected function processRepositories()
 
 		if (!method_exists($builder, 'findByType')) {
 			foreach ($this->configuredManagers as $managerName => $_) {
-				$builder->getDefinition($this->prefix($managerName . '.repositoryFactory'))
-					->addSetup('setServiceIdsMap', [[], $this->prefix('repositoryFactory.' . $managerName . '.defaultRepositoryFactory')]);
+				/** @var Nette\DI\ServiceDefinition $def */
+				$def = $builder->getDefinition($this->prefix($managerName . '.repositoryFactory'));
+				$def->addSetup('setServiceIdsMap', [[], $this->prefix('repositoryFactory.' . $managerName . '.defaultRepositoryFactory')]);
```

Comment:
why not chain?

Reply:
Because then it fails on phpstan.
getDefinition() returns ANY Definition. But the method addSetup() is defined only for ServiceDefinition. I have to tell phpstan, that $def is of type ServiceDefinition first and then call ->addSetup.
Otherwise, phpstan returns error.

Reply:
Then how about adding this?

```
private function getServiceDefinition(ContainerBuilder $builder, string $name): ServiceDefinition
{
    $definition = $builder->getDefinition($name);
    assert($definition instanceof ServiceDefinition);

	return $definition;
}
```

Reply:
good idea, fixed